pull_request_package/lib/github_status_reporter.rb

```diff
@@ -3,22 +3,57 @@
 
 class GitHubStatusReporter
   include ActiveModel::Model
-  attr_accessor :client, :logger, :pull_request
+  attr_accessor :client, :logger, :package
   
   def report
-    client.create_status('openSUSE/open-build-service', 'f73658fa927b90a0a3d28b79866215210b856390', 'pending', options) if [:success, :failed].include?(state)
+    summary = self.summary
+    state = :success
+    if summary[:failure] > 0
+      state = :failure
+    elsif summary[:pending] > 0 or summary[:success] == 0
+      state = :pending
+    end
+    count = summary[:success] + summary[:failure] + summary[:pending]
+    case state
+    when :failure
+       descr = "#{summary[:failure]}/#{count} failed"
+    when :success
+       descr = "#{count} succeeded"
+    else
+       descr = "#{summary[:pending]}/#{count} building"
+    end
+    client.create_status('openSUSE/open-build-service', 'f73658fa927b90a0a3d28b79866215210b856390', state, options.merge(description: descr))
   end
   
   def options
     options = { 
-      context: 'OBS Package build result', 
+      context: "OBS Package build result #{package.pull_request.number}",
       target_url: package.url,
     }
   end
   
-  def state
-    result = `osc api /status/project/#{package.obs_project_name}`
-    # TODO: Parse result to get the state
-    nil
+  def judge_code(code)
+    case code
+    when 'succeeded', 'excluded', 'disabled'
+      :success
+    when 'broken', 'failed', 'unresolvable'
+      :failure
+    when 'building', 'dispatching', 'scheduled', 'finished', 'blocked'
+      :pending
+    else
+      puts "UNMAP '#{code}'"
```

Comment:
logger.error("Unmapped status result #{code} in #{package.obs_package_name} ")

Reply:
Yeah, but I sure hope they are limited :)